include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -0,0 +1,408 @@
+    // Complex needs an extra sweep to have a real tridiagonal matrix.
+    return is_complex ? size - 1 : size - 2;
+  }
+
+  static SizeType nrStepsForSweep(SizeType sweep, SizeType size, SizeType band) noexcept {
+    // Sweep size-2 should be handled differently.
+    // Note: It is executed only for complex types (see nrSweeps)
+    return sweep == size - 2 ? 1 : util::ceilDiv(size - sweep - 2, band);
+  }
+
+  // Local implementation of bandToTridiag.
+  static ReturnTridiagType<T, Device::CPU> call_L(const SizeType b,
+                                                  Matrix<const T, Device::CPU>& mat_a) noexcept {
+    using MatrixType = Matrix<T, Device::CPU>;
+    using ConstTileType = typename MatrixType::ConstTileType;
+    using common::internal::vector;
+    using common::Pipeline;
+    using common::PromiseGuard;
+    using util::ceilDiv;
+
+    using hpx::unwrapping;
+    using hpx::resource::get_num_threads;
+
+    auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+
+    // note: A is square and has square blocksize
+    SizeType size = mat_a.size().cols();
+    SizeType nrtile = mat_a.nrTiles().cols();
+    SizeType nb = mat_a.blockSize().cols();
+
+    // Need share pointer to keep the allocation until all the tasks are executed.
+    auto a_ws = std::make_shared<BandBlock<T>>(size, b);
+
+    Matrix<BaseType<T>, Device::CPU> mat_trid({2, size}, {2, nb});
+    Matrix<T, Device::CPU> mat_v({size, size}, {nb, nb});
+    const auto& dist_v = mat_v.distribution();
+
+    if (size == 0) {
+      return {std::move(mat_trid), std::move(mat_v)};
+    }
+
+    const auto max_deps_size = ceilDiv(size, b);
+    vector<hpx::shared_future<void>> deps;
+    deps.reserve(max_deps_size);
+
+    auto copy_diag = [a_ws](SizeType j, const ConstTileType& source) { a_ws->copyDiag(j, source); };
+
+    auto copy_offdiag = [a_ws](SizeType j, const ConstTileType& source) {
+      a_ws->copyOffdiag(j, source);
+    };
+
+    // Copy the band matrix
+    for (SizeType k = 0; k < nrtile; ++k) {
+      hpx::shared_future<void> sf =
+          hpx::dataflow(executor_hp, unwrapping(copy_diag), k * nb, mat_a.read(GlobalTileIndex{k, k}));
+      if (k < nrtile - 1) {
+        for (int i = 0; i < nb / b - 1; ++i) {
+          deps.push_back(sf);
+        }
+        sf = hpx::dataflow(executor_hp, unwrapping(copy_offdiag), k * nb,
+                           mat_a.read(GlobalTileIndex{k + 1, k}), sf);
+        deps.push_back(sf);
+      }
+      else {
+        while (deps.size() < max_deps_size) {
+          deps.push_back(sf);
+        }
+      }
+    }
+
+    // Maximum size / (2b-1) sweeps can be executed in parallel.
+    const auto max_workers =
+        std::min(ceilDiv(size, 2 * b - 1), to_signed<SizeType>(get_num_threads("default")));
+    vector<Pipeline<SweepWorker<T>>> workers;
+    workers.reserve(max_workers);
+    for (SizeType i = 0; i < max_workers; ++i)
+      workers.emplace_back(SweepWorker<T>(size, b));
```

Comment:
```suggestion
    vector<Pipeline<SweepWorker<T>>> workers(max_workers, SweepWorker<T>(size, b));
```

Reply:
Pipeline<SweepWorker<T>> is not copyable, therefore the constructor you suggest cannot be used.

Reply:
It should, however, probably be `emplace_back(size, b)` with the loop version?

Reply:
It is a `vector<Pipeline<SweepWorker>>`, which is something I missed at the beginning 😄, so I would say no.

Reply:
Ah yes, so did I...!